src/utils/subjectPicker.js

```diff
@@ -1,46 +1,28 @@
-import pointsCalculator from './pointsCalculator'
-import { primarySubjects } from '../constants/SubjectTypes'
+import getSubjectScore from './getSubjectScore'
 
-const subjectPickerUtil = (subjects, subjectsType) => {
+const subjectPicker = (subjects, subjectsType) => {
 
-  // Filtruje przedmioty do danych przez subjectsType i aktywnych
-  let subjectsArray = subjects.filter(subject => subject.active && (subject.type === subjectsType))
-
-  if (subjectsArray.length === 0) return null
-  else {
-    let maxScoreWrapper = {
-      subject: null,
-      computedScore: null,
-      formula: null
-    }
-
-    // Zmienna odpowiedzialna za przechowywanie najlepszego przedmiotu,
-    // na poczatku znajduje sie w niej pierwszy mozliwy przedmiot
-    let maxScoreSubject = subjectsArray[0]
-
-    // Zmienna przechowuje obliczony wynik + rodzaj rownania (najlepszego przedmiotu)
-    let maxScoreObj = pointsCalculator(maxScoreSubject.elementaryScore, maxScoreSubject.advancedScore)
-
-    // Przechodze przez tablice przediotow w poszukiwaniu najbardziej wartosciowego
-    subjectsArray.forEach((subject) => {
-      let currentScoreObj = pointsCalculator(subject.elementaryScore, subject.advancedScore)
-
-      if (currentScoreObj.computedScore > maxScoreObj.computedScore) {
-        // Zapisuje go do zmiennych
-        maxScoreSubject = subject
-        maxScoreObj = currentScoreObj
-      }
-    })
-    // Przepisuje do wrappera przedmiot, jego punkty oraz wzor zgodnie z ktorym zostal wyliczony maksymalny wynik
-    maxScoreWrapper.subject = maxScoreSubject
-
-    if (subjectsType === primarySubjects) maxScoreWrapper.computedScore = maxScoreObj.computedScore * 4
-    else maxScoreWrapper.computedScore = maxScoreObj.computedScore
+  let maxScoreWrapper = {
+    subject: {},
+    computedScore: 0,
+    formula: '',
+    isEmpty: false
+  }
 
-    maxScoreWrapper.formula = maxScoreObj.formula
+  const subjectsArray = subjects.filter(subject => subject.active && (subject.type === subjectsType))
 
+  if (!subjectsArray.length) {
+    maxScoreWrapper = { ...maxScoreWrapper, isEmpty: true }
     return maxScoreWrapper
   }
+
+  return subjectsArray.reduce((currentMaxWrapper, subject) => {
+    const currentScoreObj = getSubjectScore(subject)
+
+    return (currentScoreObj.computedScore > currentMaxWrapper.computedScore)
+      ? { subject, ...currentScoreObj, isEmpty: false }
```

Comment:
Order of these elements was made intentionally. I knew that there is no override and wanted to preserver the default order. BTW I think I can't get rid of `isEmpty` in the 23nd line, because newly created object lacks this property and errors appear.